internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +945,37 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(ctx context.Context,
+	mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get the state of replication
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		log.ErrorLog(ctx, err.Error())
+
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	localState := localStatus.GetState()
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if isUp {
+		switch localState {
+		case librbd.MirrorImageStatusStateUnknown.String():
+			return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "mirroring is in unknown state", nil
+		case librbd.MirrorImageStatusStateError.String():
+			return replication.GetVolumeReplicationInfoResponse_ERROR, "mirroring is in error state", nil
+		case librbd.MirrorImageStatusStateReplaying.String():
+			return replication.GetVolumeReplicationInfoResponse_HEALTHY, "image/group is secondary", nil
+		case librbd.MirrorImageStatusStateStopped.String():
+			return replication.GetVolumeReplicationInfoResponse_HEALTHY, "image/group is primary", nil
+		default:
+			return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "mirroring is in intermediatory state", nil
+		}
+	}
+
+	return replication.GetVolumeReplicationInfoResponse_DEGRADED, "Mirroring is down", nil
```

Comment:
set own message from cephcsi only if we dont find any message in the description.